pkg/k8shandler/clusterrestart.go

```diff
@@ -0,0 +1,552 @@
+package k8shandler
```

Comment:
This file represents essentially the check/update state machinery for rolling restarts and cert redeploys. Can you add some unit tests to ensure the state machine does what is expected to do.